### PR TITLE
Build: Include JSON files in zip archive

### PR DESCRIPTION
**Pull request by ockham**
## Description

Tentative fix for #19770.

In the zip archive built by `bin/build-plugin-zip.sh`, include JSON files found in the `build/` directory.

Question: Is this casting too wide a net, i.e. including JSON files that we don't want/need to include?

## How has this been tested?
Hasn't yet :grimacing: 

## Types of changes

See above.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR. .

### CODE REVIEW

File: bin/build-plugin-zip.sh [hunk 1]
```diff
@@ -109,7 +109,7 @@ npm run build
 php bin/generate-gutenberg-php.php > gutenberg.tmp.php
 mv gutenberg.tmp.php gutenberg.php
 
-build_files=$(ls build/*/*.{js,css,asset.php} build/block-library/blocks/*.php)
+build_files=$(ls build/*/*.{js,json,css,asset.php} build/block-library/blocks/*.php)
```
Comment (koke) [inline bin/build-plugin-zip.sh]: I was looking at a fix for this and went with this instead, which is a bit more specific:

```suggestion
build_files=$(ls build/*/*.{js,css,asset.php} build/block-library/blocks/*.{php,json})
```

Right now, it doesn't seem there are any other JSON files though:

```
$ find build -name '*.json'                           
build/block-library/blocks/shortcode.json
```

**Review by koke — CHANGES_REQUESTED** [review]
I tried running `npm run package-plugin`:

```
ls: build/*/*.json: No such file or directory
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! gutenberg@7.2.0 package-plugin: `./bin/build-plugin-zip.sh`
npm ERR! Exit status 1
```

The file list is a shell expansion and not a glob pattern, and there's no JSON files in `build/*`, since they are a level deeper.

The one I suggested is tested and working

**Comment by ockham** [comment]
Thanks @koke, going with yours!

**Review by koke — APPROVED** [review]
👍 Thanks for taking care of this

**Comment by gziolo** [comment]
@ockham - thanks for fixing it!

**Comment by gziolo** [comment]
@ellatrix - this patch needs to be cherry-picked for the 7.3 release.

**Pull request merged**